src/lib/consensus/proof_of_signature.ml

```diff
@@ -175,16 +188,16 @@ module Make (Inputs : Inputs_intf) :
 
   let block_interval_ms = Time.Span.to_ms proposal_interval
 
-  let generate_transition ~previous_protocol_state ~blockchain_state
-      ~local_state:_ ~time:_ ~keypair ~transactions:_ ~snarked_ledger_hash:_
-      ~supply_increase:_ ~logger:_ =
+  let generate_transition ~previous_protocol_state ~blockchain_state ~time:_
+      ~proposal_data ~transactions:_ ~snarked_ledger_hash:_ ~supply_increase:_
+      ~logger:_ =
     let previous_consensus_state =
       Protocol_state.consensus_state previous_protocol_state
     in
     (* TODO: sign protocol_state instead of blockchain_state *)
     let consensus_transition_data =
-      Consensus_transition_data.create_value
-        ~private_key:keypair.Signature_lib.Keypair.private_key blockchain_state
+      Consensus_transition_data.create_value ~private_key:proposal_data
```

Comment:
just to eliminate arguments -- I think it makes sense conceptually too since generically there's no reason a consensus mechanism should need a key pair (consider eg PoW)

Reply:
PoW doesn't need a keypair for `next_proposal` either, but I get your point.